native/cocos/editor-support/spine-wasm/spine-type-export.cpp

```diff
@@ -85,20 +34,467 @@ using SPVectorPathConstraintPtr = Vector<PathConstraint*>;
+        static constexpr TYPEID get() { \
+            return #cls; \
+        } \
+    }; \
+    template<> \
+    struct LightTypeID<const cls&> { \
+        static constexpr TYPEID get() { \
+            return #cls; \
+        } \
+    }; \
+}}
+
+#define GETTER_RVAL_TO_PTR(ClassType, Method, ReturnType) \
+    optional_override([](ClassType &obj) { return const_cast<ReturnType>(&obj.Method()); })
+
+
+DEFINE_SPINE_CLASS_TYPEID(ConstraintData)
+DEFINE_SPINE_CLASS_TYPEID(IkConstraintData)
+DEFINE_SPINE_CLASS_TYPEID(PathConstraintData)
+DEFINE_SPINE_CLASS_TYPEID(Attachment)
+DEFINE_SPINE_CLASS_TYPEID(VertexAttachment)
+DEFINE_SPINE_CLASS_TYPEID(BoundingBoxAttachment)
+DEFINE_SPINE_CLASS_TYPEID(ClippingAttachment)
+DEFINE_SPINE_CLASS_TYPEID(MeshAttachment)
+DEFINE_SPINE_CLASS_TYPEID(PathAttachment)
+DEFINE_SPINE_CLASS_TYPEID(PointAttachment)
+DEFINE_SPINE_CLASS_TYPEID(RegionAttachment)
+DEFINE_SPINE_CLASS_TYPEID(AttachmentLoader)
+DEFINE_SPINE_CLASS_TYPEID(AtlasAttachmentLoader)
+DEFINE_SPINE_CLASS_TYPEID(Interpolation)
+DEFINE_SPINE_CLASS_TYPEID(PowInterpolation)
+DEFINE_SPINE_CLASS_TYPEID(PowOutInterpolation)
+DEFINE_SPINE_CLASS_TYPEID(Updatable)
+DEFINE_SPINE_CLASS_TYPEID(IkConstraint)
+DEFINE_SPINE_CLASS_TYPEID(PathConstraint)
+DEFINE_SPINE_CLASS_TYPEID(TransformConstraintData)
+DEFINE_SPINE_CLASS_TYPEID(TransformConstraint)
+DEFINE_SPINE_CLASS_TYPEID(Bone)
+DEFINE_SPINE_CLASS_TYPEID(Timeline)
+DEFINE_SPINE_CLASS_TYPEID(CurveTimeline)
+DEFINE_SPINE_CLASS_TYPEID(TranslateTimeline)
+DEFINE_SPINE_CLASS_TYPEID(ScaleTimeline)
+DEFINE_SPINE_CLASS_TYPEID(ShearTimeline)
+DEFINE_SPINE_CLASS_TYPEID(RotateTimeline)
+DEFINE_SPINE_CLASS_TYPEID(ColorTimeline)
+DEFINE_SPINE_CLASS_TYPEID(TwoColorTimeline)
+DEFINE_SPINE_CLASS_TYPEID(AttachmentTimeline)
+DEFINE_SPINE_CLASS_TYPEID(DeformTimeline)
+DEFINE_SPINE_CLASS_TYPEID(EventTimeline)
+DEFINE_SPINE_CLASS_TYPEID(DrawOrderTimeline)
+DEFINE_SPINE_CLASS_TYPEID(IkConstraintTimeline)
+DEFINE_SPINE_CLASS_TYPEID(TransformConstraintTimeline)
+DEFINE_SPINE_CLASS_TYPEID(PathConstraintPositionTimeline)
+DEFINE_SPINE_CLASS_TYPEID(PathConstraintMixTimeline)
+DEFINE_SPINE_CLASS_TYPEID(VertexEffect)
+DEFINE_SPINE_CLASS_TYPEID(JitterVertexEffect)
+DEFINE_SPINE_CLASS_TYPEID(SwirlVertexEffect)
+
+
+DEFINE_ALLOW_RAW_POINTER(BoneData)
+DEFINE_ALLOW_RAW_POINTER(Bone)
+DEFINE_ALLOW_RAW_POINTER(Slot)
+DEFINE_ALLOW_RAW_POINTER(SlotData)
+DEFINE_ALLOW_RAW_POINTER(VertexAttachment)
+DEFINE_ALLOW_RAW_POINTER(Color)
+DEFINE_ALLOW_RAW_POINTER(EventData)
+DEFINE_ALLOW_RAW_POINTER(Skeleton)
+DEFINE_ALLOW_RAW_POINTER(SkeletonData)
+DEFINE_ALLOW_RAW_POINTER(Skin)
+DEFINE_ALLOW_RAW_POINTER(Animation)
+DEFINE_ALLOW_RAW_POINTER(AnimationStateData)
+DEFINE_ALLOW_RAW_POINTER(TrackEntry)
+DEFINE_ALLOW_RAW_POINTER(IkConstraintData)
+DEFINE_ALLOW_RAW_POINTER(PathConstraintData)
+DEFINE_ALLOW_RAW_POINTER(TransformConstraintData)
+DEFINE_ALLOW_RAW_POINTER(SPVectorUnsignedShort)
+DEFINE_ALLOW_RAW_POINTER(SPVectorFloat)
```

Comment:
GETTER in `.property(...)` doesn't support a function that returns `raw pointer`, so we use `DEFINE_ALLOW_RAW_POINTER` to make embind core code know that we need to return `raw pointer` for the getter.